sdk/core/azure-core/inc/azure/core/base64.hpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Utility functions to help convert between binary data and UTF-8 encoded text that is
+ * represented in base 64.
+ */
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+namespace Azure { namespace Core {
```

Comment:
What about making this API part of Azure::core::string?

Reply:
Hmm. Maybe. I don't feel strongly either way. @Jinming-Hu, @JeffreyRichter what do you think? Should Base64 APIs be in `Azure::Core` or `Azure::Core::Strings`?

Reply:
Well, I don't think so. I feel encoding/decoding stuff is more related to MD5 which you'll move into `Azure::Core` later. I guess all of you will agree that MD5 should be part of `Azure::Core::Strings`.

Reply:
> Both base64 and MD5 actually handle binary data, and string is kind of different from binary data. In my mind, string is usually something that's printable.

True, except base64 encoding **does** convert binary data into printable text, using only ASCII characters [A-z][a-z] and a couple of symbols.

Reply:
Do we need a PUBLIC Azure::Core:Strings namespace at all? I see it has only 3 function in it:
   bool LocaleInvariantCaseInsensitiveEqual(const std::string& lhs, const std::string& rhs) noexcept;
   std::string const ToLower(const std::string& src) noexcept;
   unsigned char ToLower(const unsigned char src) noexcept;

Why would a customer need to ask the Azure SDK to lowercase a string or compare for equality?

Regardless of whether we keep the Strings namespace or not, I think the base64 functions should NOT go in that namespace.

Reply:
> Do we need a PUBLIC Azure::Core:Strings namespace at all?

Addressed in https://github.com/Azure/azure-sdk-for-cpp/pull/1337

> Regardless of whether we keep the Strings namespace or not, I think the base64 functions should NOT go in that namespace.

OK, marking as resolved.

Reply:
@ahsonkhan as I understand it, what Jeffrey suggested is that we do not need `Strings` namespace at all. But you just moved it into `Internal`, it's still there. Should we just remove the namespace?

Reply:
I don't think we should remove it.
I would like not to have any class/API directly exposed from `::interal` but always under some subclassification like `::interal::strings` or `::interna::http`.

To me, it looks better and prevents ::internal from eventually exposing a long list of APIs.

Probably the good news could be that the internal layer, considered as implementation, could be changed even after GA. So, we could keep updating/improving this layer

Reply:
We had a discussion in the linked PR about keeping the sub-namespace within `internal`, at which point (as Victor mentioned) it becomes implementation detail and shouldn't matter anymore to an end user:
https://github.com/Azure/azure-sdk-for-cpp/pull/1337#discussion_r556141538